test/node/bson_undefined.test.ts

```diff
@@ -0,0 +1,106 @@
+import { expect } from 'chai';
+import { bufferFromHexArray } from './tools/utils';
+import { BSON } from '../register-bson';
+import { BSON_DATA_NULL } from '../../src/constants';
+
+describe('BSON undefined', () => {
+  const KEY_A = '6100';
+  const KEY_0 = '3000';
+  const KEY_1 = '3100';
+  const KEY_2 = '3200';
+  const bsonDocWithUndefined = bufferFromHexArray([
+    '06', // BSON undefined
+    KEY_A
+  ]);
+
+  describe('when deserialize is given BSON bytes with undefined value', function () {
+    it('returns a javascript undefined value', () => {
+      const doc = BSON.deserialize(bsonDocWithUndefined);
+      expect(doc).to.have.own.property('a').that.is.undefined;
+    });
+  });
+
+  describe('when serialize is given a javascript object that contains undefined', () => {
+    describe('when ignoreUndefined is set to false', function () {
+      it('serializes to document with a set to BSON null (type=10)', () => {
+        const jsObject = BSON.deserialize(bsonDocWithUndefined);
```

Comment:
```suggestion
        const jsObject = { a: undefined };
```

Is there a reason we can't use an object literal?

Reply:
good point, fixed